### PR TITLE
[MS] Added an endpoint for parsing a handson table as html and return…

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
Added an endpoint for parsing a handson table as html and returning valid json (not implemented yet).
Also moved html utilities into their own package.

### CODE REVIEW

File: api/parse.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	setContentType(w, contentJSON)
+	if err != nil {
+		log.Error(err, log.Data{"parse_request": parseRequest})
+		setErrorCode(w, err)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(bytes)
+	if err != nil {
```
Comment (daiLlew) [inline api/parse.go]: If you don't care about the result you can eliminate the extra line i.e. 
`if _, err = w.Write(bytes); err != nil {
log.Error(err, log.Data{"parse_request": parseRequest})
setErrorCode(w, err)
return
}`
But I won't not approve the review just for that

File: htmlutil/htmlutil.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	return &html.Node{Type: html.TextNode, Data: text}
+}
+
+// GetAttribute finds an attribute for the node - returns empty string if not found
+func GetAttribute(node *html.Node, key string) string {
+	for _, attr := range node.Attr {
+		if attr.Key == key {
+			return attr.Val
+		}
+	}
```
Comment (daiLlew) [inline htmlutil/htmlutil.go]: is it worth logging here the attribute could not be found?

Comment (YetAnotherMatt) [reply htmlutil/htmlutil.go]: GetAttribute is used by HasAttributes, so you'd end up with a *lot* of debug output

File: api/parse.go (hunk 1 above)
Comment (YetAnotherMatt) [reply api/parse.go]: so
```
	_, err = w.Write(bytes); 
	if err != nil {
		log.Error(err, log.Data{"parse_request": parseRequest})
		setErrorCode(w, err)
		return
	}

```
becomes:
```
	_, err = w.Write(bytes);  if err != nil {
		log.Error(err, log.Data{"parse_request": parseRequest})
		setErrorCode(w, err)
		return
	}
```

**Pull request merged**